mesonbuild/dependencies/base.py

```diff
@@ -1329,8 +1343,11 @@ def _detect_dep(self, name: str, modules: List[str], args: List[str]):
         compileDefinitions = []
         compileOptions = []
         libraries = []
-        for i in modules:
+        for i, required in modules:
             if i not in self.targets:
+                if not required:
+                    mlog.warning('CMake: Optional CMake target', mlog.bold(i), 'for', mlog.bold(name), 'was not found')
+                    continue
```

Comment:
There's an api mismatch between this and the config tool version. The config tool targets are all lowercase while the cmake ones are camel case. We could just store a mapping between the lowered form and the camel case form that cmake uses and convert as necessary, unless you have a better idea.